File: OOP/Mixins.py
import json
from datetime import datetime, timezone


class SerializableMixin:
    def serialize(self):
        return json.dumps(self.__dict__)

    def deserialize(self, json_text):
        self.__dict__.update(json.loads(json_text))
        return self


class TimestampedMixin:
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.created_at = datetime.now(timezone.utc).isoformat()


class User:
    def __init__(self, name):
        self.name = name


class EnhancedUser(SerializableMixin, TimestampedMixin, User):
    pass


u = EnhancedUser("Nina")
print(u.serialize())

File: OOP/Mixins.js
// Mixins simulate interfaces — JavaScript has no native interface keyword
const Serializable = (Base) => class extends Base {
  serialize()   { return JSON.stringify(this); }
  deserialize(json) { return Object.assign(this, JSON.parse(json)); }
};

const Timestamped = (Base) => class extends Base {
  constructor(...args) {
    super(...args);
    this.createdAt = new Date().toISOString();
  }
};

class User { constructor(name) { this.name = name; } }

class EnhancedUser extends Serializable(Timestamped(User)) {}

const u = new EnhancedUser("Nina");
console.log(u.serialize());  // JSON with name and createdAt